Computed properties, observers and simple properties

Shadowere opened the pull request:
- Adds computed properties to model-proxy;
- Adds observers to model-proxy;
- Adds simple-properties to model-proxy;
- Updates package.json version;

Shadowere committed:
Computed properties, observers and simple properties

- Adds computed properties to model-proxy;
- Adds observers to model-proxy;
- Adds simple-properties to model-proxy;
- Updates package.json version;

Oreoz commented on addon/services/model-proxy.js:
```diff
@@ -12,24 +14,145 @@ export default Service.extend({
+    internalProxy.currentState = currentState;
+    internalProxy._internalModel.currentState = currentState;
+  },
+  _addObserver(proxy, model, propertyName) {
```
I think the spacing between functions in this file lacks consistency -- either single space between functions or no spaces (I'm in the space camp).

Shadowere committed:
Code-climate fixes

- Adds consistent spacing;
- Fixes code-climate issues (< 25 lines per function);

Shadowere committed:
Model-Extractor service

- Extracts the 'gets' for observers, computed properties and such in a model-extractor service;
